### PR TITLE
Working on support for managed_ptr in HIP

### PR DESCRIPTION
**Pull request by adayton1**
* Fixed some bugs in managed_ptr
* Added reproducers for what appear to be rocm bugs

### CODE REVIEW

**Review by robinson96 — DISMISSED** [review]
LGTM. Thanks for pushing on this.

File: examples/managed_ptr_example.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  // Create an object on the host, and if either CUDA or HIP is enabled,
+  // on the device as well
+  chai::managed_ptr<TestBase> derived = chai::make_managed<TestDerived>(42);
+
+  // chai::managed_ptr can be accessed on the host
+  forall(sequential(), 0, 1, [=] CHAI_HOST (int i) {
+    printf("Result of virtual function call on host: %d\n", derived->getValue());
+  });
+
+#if defined(CARE_GPUCC)
```
Comment (davidbeckingsale) [inline examples/managed_ptr_example.cpp:39]: Does this macro need changing?

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]

File: examples/managed_ptr_example.cpp (hunk 1 above)
Comment (adayton1) [reply examples/managed_ptr_example.cpp]: Right you are! Fixed!

**Review by davidbeckingsale — APPROVED** [review]

**Review by robinson96 — APPROVED** [review]
LGTM

**Pull request merged**